includes/class-admin.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * Plugin Name:       Coding Challenge
+ * 
+ * @package Coding_Challenge
+ * @subpackage Coding_Challenge_main_class
+ * @since 0.1.0 ( plugin creation)
+ * @author mikesm1118
+ * 
+ * @wordpress-plugin 
+ * Plugin Name:       Coding Challenge
+ * Plugin URI:        https://wordpress.org/plugins/coding-challenge
+ * Description:       This plugin creates an admin menu for changing the website's background color.
+ * Version:           0.1.0
+ * Requires at least: 5.1
+ * Requires PHP:      5.6
+ * Author:            mikesm1118
+ * Author URI:        https://profiles.wordpress.org/mikesm1118/
+ * License:           GPL v2 or later
+ * License URI:       https://www.gnu.org/licenses/gpl-2.0.html
+ * Text Domain:       coding-challenge
+ * Domain Path:       /languages
+ */
+
+/**
+ * Class:  Coding_Challenge_Admin
+ * 
+ * Main class and activation hooks. 
+ *
+ * @link https://wordpress.org/plugins/coding-challenge
+ *
+ * @package Coding_Challenge
+ * @subpackage Coding_Challenge_Admin_menu
+ * @since 0.1.0 ( plugin creation)
+ */
+class Coding_Challenge_Admin {
+
+	/** 
+	 * Constructor.
+	 * 
+	 * @since 0.1.0
+	 * 
+	 * @param Coding_Challenge $coding_challenge coding challenge object.
+	 */
+	public function __construct( $coding_challenge ) {
+		$this->coding_challenge = $coding_challenge;
+	}
+
+	/** 
+	 * Register admin menu.
+	 * 
+	 * Adds the plugin to the admin dashboard.
+	 * 
+	 * @since 0.1.0
+	 */
+	public function register_admin_menu() {
+		add_menu_page(
+			'Coding Challenge',
+			'Coding Challenge',
+			'manage_options',
+			'coding-challenge',
+			array( $this, 'display_admin_page' ),
+			'',
+			65
+		);
+	}
+	
+	/** 
+	 * Register admin settings.
+	 * 
+	 * Adds the plugin features to the admin menu
+	 * 
+	 * @since 0.1.0
+	 * 
+	 */
+	public function register_admin_settings() {
+		register_setting(
+			'background_setting',
+			'background_setting',
+			''
+		);
+
+	// Add section to options settings.
```

Comment:
These comments are incorrectly indented. You have this:

```
	// Add section to options settings.
		add_settings_section(
			'setting_section',
			'Background Settings',
			array( $this, 'display_section' ),
			'coding-challenge'
		);
```

When you should have this:

```
		// Add section to options settings.
		add_settings_section(
			'setting_section',
			'Background Settings',
			array( $this, 'display_section' ),
			'coding-challenge'
		);
```